modules/prebidServerBidAdapter/index.js

```diff
@@ -861,6 +861,33 @@ Object.assign(ORTB2.prototype, {
       request.ext.prebid = mergeDeep(request.ext.prebid, s2sConfig.extPrebid);
     }
 
+    // get reference to pbs config schain bidder names (if any exist)
+    const pbsSchainBidderNamesArr = request.ext.prebid?.schains ? request.ext.prebid.schains.flatMap(s => s.bidders) : [];
+    // create an schains object
+    const schains = Object.fromEntries(
+      (request.ext.prebid?.schains || []).map(({bidders, schain}) => [JSON.stringify(schain), {bidders: new Set(bidders), schain}])
+    );
+
+    // compare bidder specific schains with pbs specific schains
+    request.ext.prebid.schains = Object.values(
+      bidRequests
+        .map((req) => [req.bidderCode, req.bids[0].schain])
+        .reduce((chains, [bidder, chain]) => {
+          const chainKey = JSON.stringify(chain);
+          if (chainKey && !chains.hasOwnProperty(chainKey)) {
+            if (pbsSchainBidderNamesArr.indexOf(bidder) !== -1) {
+              logInfo(`bidder-specific schain for ${bidder} skipped due to existing entry`);
+            } else {
+              chains[chainKey] = {bidders: new Set(), schain: chain};
+              chains[chainKey].bidders.add(bidder);
```

Comment:
I believe the `bidders.add(bidder)` needs to be detented out of `if (!chains.hasOwnProperty(chainKey))`; otherwise, if multiple bidders have chains that serialize to the same key, you are losing the association for every one of them except the first. 

This would also be a good test case, if I understand your intent, you want two separate bid requests with identical chains to be "merged" into a single entry in ext.prebid.schains.

Reply:
adjusted the logic for this and also added a test case